codec/codec.go

```diff
@@ -0,0 +1,46 @@
+package codec
```

Comment:
I'm not sure this warrants its own package. It also leads to a clumsy name (`codec.Codec`). Can it not go in the `grpc` package?